### PR TITLE
Add search to catalog

### PR DESCRIPTION
**Pull request by spadgett**
Basic search support, with little styling.

screen shot 2017-04-03 at 4 20 12 pm

TODO:

- [ ] Style the search results (@cshinn)
- [x] Support image URLs, not just font icons
- [x] Show tags for builder images
- [x] Use KeywordsService (currently broken with mock data)
- [x] Defer search when user starts typing before items are loaded
- [ ] See if we can have one overlay-panel instead of having to duplicate the logic. Currently it's tied to the services view.
- [ ] Add unit tests

### CODE REVIEW

**Comment by spadgett** [comment]
I hate having multiple WIP in flight, but I know @cshinn was going to look at styling the search control.

**Comment by spadgett** [comment]
@dtaylor113 I checked GitHub markdown syntax, and I didn't see any easy way to make the "not ready for code review" text red :)

**Comment by spadgett** [comment]
cc @serenamarie125

**Comment by pmorie** [comment]
ship it

**Comment by spadgett** [comment]
@jwforres @dtaylor113 This is probably far enough along for a review.

@jwforres I didn't debounce the search because it didn't feel as good. There's a `typeahead-wait-ms` setting on the typeahead control if we want it.

File: src/components/catalog-search/catalog-search.controller.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      items = items.concat(_.map(this.ctrl.imageStreams, (imageStream) => {
+        return this.Catalog.getImageItem(imageStream);
+      }));
+
+      // Remove null items (non-builder images).
+      items = _.reject(items, (item) => {
+        return !item;
+      });
+
+      this.ctrl.allItems = _.sortBy(items, 'name');
```
Comment (jwforres) [inline src/components/catalog-search/catalog-search.controller.ts]: is it possible for someone to start typing into the filter while the service class or image streams are still loading? do you need to run filter here

**Comment by spadgett** [comment]
> is it possible for someone to start typing into the filter while the service class or image streams are still loading? do you need to run filter here

Added a second commit that defers the search until items load.

**Comment by spadgett** [comment]
Squashed and rebased

**Comment by jwforres** [comment]
can merge once travis passes

**Pull request merged**